### PR TITLE
[Error Handling 2] - Error Codes

### PR DESCRIPTION
**Pull request by W3stside**
Adds quote error handling codes for `Insufficient Liquidity` and `FeeExceedsPrice` (<-- not from backend)

Waterfalls into #745

### CODE REVIEW

File: src/custom/utils/operator/error.ts [hunk 1]
```diff
@@ -8,29 +13,23 @@ export enum ApiErrorCodes {
   InsufficientFunds = 'InsufficientFunds',
   InsufficientFee = 'InsufficientFee',
   UnsupportedToken = 'UnsupportedToken',
-  WrongOwner = 'WrongOwner'
+  WrongOwner = 'WrongOwner',
+  NotFound = 'NotFound',
```
Comment (anxolin) [inline src/custom/utils/operator/error.ts:17]: Is this something the backend returns?
This is just an enum with backend codes https://github.com/gnosis/gp-v2-services/blob/0bd5f7743bebaa5acd3be13e35ede2326a096f14/orderbook/openapi.yml#L562

File: src/custom/utils/operator/error.ts [hunk 2]
```diff
@@ -57,9 +56,11 @@ export default class OperatorError extends Error {
       return OperatorError.apiErrorDetails.UNHANDLED_ERROR
     }
   }
-  static async getErrorForUnsuccessfulPostOrder(response: Response) {
+
+  static async getErrorFromStatusCode(response: Response) {
     switch (response.status) {
       case 400:
+      case 404:
```
Comment (anxolin) [inline src/custom/utils/operator/error.ts:63]: Will this work? Afaik `NotFound` is not a valid error in the backend, so it won't be handled...

File: src/custom/utils/operator/error.ts (hunk 1 above)
Comment (alfetopito) [reply src/custom/utils/operator/error.ts:17]: This is probably the new state added the other day https://github.com/gnosis/gp-v2-services/blob/b141b2bf32a0c861ef952d58290161abc3265b24/orderbook/src/fee.rs#L88

Guess they forgot to update open api docs

File: src/custom/utils/operator/error.ts [hunk 3]
```diff
@@ -8,29 +13,23 @@ export enum ApiErrorCodes {
...
+  InvalidSignature = 'The order signature is invalid. Check whether your Wallet app supports off-chain signing.',
+  MissingOrderData = 'The order has missing information',
+  InsufficientValidTo = 'The order you are signing is already expired. This can happen if you set a short expiration in the settings and waited too long before signing the transaction. Please try again.',
+  InsufficientFunds = "The account doesn't have enough funds",
+  UnsupportedToken = 'One of the tokens you are trading is unsupported. Please read the FAQ for more info.',
+  WrongOwner = "The signature is invalid.\n\nIt's likely that the signing method provided by your wallet doesn't comply with the standards required by CowSwap.\n\nCheck whether your Wallet app supports off-chain signing (EIP-712 or ETHSIGN).",
+  NotFound = 'Token pair selected has insufficient liquidity',
+  FeeExceedsFrom = 'Fee amount for selected pair exceeds "from" amount',
+  UNHANDLED_ERROR = 'The order was not accepted by the network'
```
Comment (alfetopito) [inline src/custom/utils/operator/error.ts:32]: oh my, the conflicts https://github.com/gnosis/cowswap/pull/737/files

:sob:

File: src/custom/utils/operator/error.ts [hunk 4]
```diff
@@ -40,7 +39,7 @@ export default class OperatorError extends Error {
 
   // Status 400 errors
   // https://github.com/gnosis/gp-v2-services/blob/9014ae55412a356e46343e051aefeb683cc69c41/orderbook/openapi.yml#L563
-  static apiErrorDetails = API_ERROR_CODE_DESCRIPTIONS
+  static apiErrorDetails = ApiErrorCodeDetails
```
Comment (alfetopito) [inline src/custom/utils/operator/error.ts:42]: After this change, do you need `static apiErrorDetails` at all? Can't you just use the enum?

**Review by alfetopito — APPROVED** [review]
Minor comments otherwise looks ok

File: src/custom/utils/operator/error.ts (hunk 2 above)
Comment (W3stside) [reply src/custom/utils/operator/error.ts:63]: it does work, though it's more a workaround until the proper error is returned. you can test it by trying what i wrote in #747 or just going here: https://protocol-rinkeby.dev.gnosisdev.com/api/v1/markets/0x5592E2C0cf4dbc12D3aB100b257153436a1f0FEa-0x6D7F0754FFeb405d23C51CE938289d4835bE3b14/buy/14882129914429537408

File: src/custom/utils/operator/error.ts (hunk 4 above)
Comment (W3stside) [reply src/custom/utils/operator/error.ts:42]: can just use enum, yep

File: src/custom/utils/operator/error.ts (hunk 3 above)
Comment (W3stside) [reply src/custom/utils/operator/error.ts:32]: uh oh

Comment (W3stside) [reply src/custom/utils/operator/error.ts:32]: meh actually should be fine, doesn't look too bad

Comment (alfetopito) [reply src/custom/utils/operator/error.ts:32]: That's ok, I just like to complain :P

File: src/custom/utils/operator/error.ts (hunk 4 above)
Comment (W3stside) [reply src/custom/utils/operator/error.ts:42]: noted in #751

**Pull request merged**